declaration.go

```diff
@@ -221,6 +227,10 @@ func (d *declarationOf[T]) construct(ctx *Context) error {
 		return nil
 	}
 
+	if sc, ok := any(&d.value).(selfConstructor); ok {
```

Comment:
Sneaky move to convert it to `any` first and then try to type-assert it to `selfConstructor`, nice 😃

Reply:
Yeah, a bit annoying and weird and dumb, but it works haha.